models/query/report-query.js

```diff
@@ -68,15 +69,19 @@ ReportQuery.prototype.toMongooseFilter = function() {
   // Determine author filter
   if (this.author) {
     filter.author = {};
-    filter.author.$in = toRegexp.alli(this.author.trim().split(/\s*,\s*/).sort());
+    filter.author.$options = 'i';
+    filter.author.$regex = this.author;
   }
   if (this.tags) {
-    filter.tags = { $all: toRegexp.allCaseInsensitive(this.tags) };
```

Comment:
Looks like the only remaining `toRegexp` usage after this is in `incident.js`, can we get rid of that too and delete the whole `toRegexp` file? Would be a nice cleanup, that file seems smelly to me.